### PR TITLE
Add ResourceQuota CRD

### LINKED ISSUE #9119: Implement Project ResourceQuotas CRD
Implement ResourceQuotas CRD that can be set per project.

### PR DESCRIPTION
**Pull request by lsviben**
**What does this PR do / Why do we need it**:
Adds the ResourceQuota CRD, which will be used to set cluster resource quotas for (currently only) projects

**Does this PR close any issues?**:
Fixes #9119 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/kubermatic/v1/resource_quota.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// QuotaSubject describes the entity to which the quota applies to.
+type QuotaSubject struct {
+	// Name of the quota subject.
+	Name string `json:"name"`
+
+	// +kubebuilder:validation:Enum=project
+	// +kubebuilder:default=project
+
+	// Type of the quota subject. For now the only possible type is project
+	Type string `json:"type"`
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/resource_quota.go]: Would "kind" maybe be a better name for this?

File: pkg/apis/kubermatic/v1/resource_quota.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	// QuotaSubject specifies to which entity the quota applies to.
+	QuotaSubject QuotaSubject `json:"quotaSubject"`
+	// Quota specifies the current maximum allowed usage of resources.
+	Quota ResourceDetails `json:"quota"`
+}
+
+// ResourceQuotaStatus describes the current state of a resource quota.
+type ResourceQuotaStatus struct {
+	// ResourceConsumption is holds the current usage of resources for all seeds.
+	ResourceConsumption ResourceDetails `json:"globalUsage,omitempty"`
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/resource_quota.go]: Field and tag names do not match, I would prefer if we renamed the field to `GlobalUsage` or `GlobalConsumption`

File: pkg/apis/kubermatic/v1/resource_quota.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ResourceQuotaSpec   `json:"spec,omitempty"`
+	Status ResourceQuotaStatus `json:"status,omitempty"`
+}
+
+// ResourceQuotaSpec describes the desired state of a resource quota.
+type ResourceQuotaSpec struct {
+	// QuotaSubject specifies to which entity the quota applies to.
+	QuotaSubject QuotaSubject `json:"quotaSubject"`
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/resource_quota.go]: Wouldn't `subject` be enough?` "ResourceQuota.quotaSubject" sounds doublespeaky.

File: pkg/apis/kubermatic/v1/resource_quota.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:generate=true
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type="date"
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/resource_quota.go:27]: We could add subject name and type to the printer columns, WDYT?

File: pkg/apis/kubermatic/v1/resource_quota.go (hunk 2 above)
Comment (lsviben) [reply pkg/apis/kubermatic/v1/resource_quota.go]: right, I was thinking about these and in the end left them mixed :D. Changed to [Local/Global]Usage.

File: pkg/apis/kubermatic/v1/resource_quota.go (hunk 4 above)
Comment (lsviben) [reply pkg/apis/kubermatic/v1/resource_quota.go:27]: good point

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**